src/Device/Driver/XCVario.cpp

```diff
@@ -158,6 +160,20 @@ XVCDevice::ParseNMEA(const char *String, NMEAInfo &info)
     return false;
 }
 
+// For documentation refer to chapter 10.1.3 Device Driver/XCVario in mulilingual handbook: https://xcvario.de/handbuch
+
+bool 
+XVCDevice::PutQNH(const AtmosphericPressure &pres, OperationEnvironment &env)
+{
+  /* the XCVario understands "!g,q<NNNN>" command for QNH updates with recent builds */
+  char buffer[32];
+  unsigned qnh = uround(pres.GetPascal()/100);
```

Comment:
Look at the implementation of `GetPascal()`:
https://github.com/XCSoar/XCSoar/blob/c3997f5458039805f9a09727a89eae11132c8084/src/Atmosphere/Pressure.hpp#L90-L92
If you want hPa, and there's a method called `GetHectoPascal()`, it's most likely the most efficient way to obtain the hectopascal value.
I'll merge this code now, but I'd also welcome a patch which optimizes this line. It would also make the code more readable!

Reply:
Btw. usually, the compiler would not be allowed to optimize this `value / 100 * 100` away into just `value`; but since XCSoar uses `-ffast-math`, we allow the compiler to do such unsafe optimizations. It's not guaranteed that this optimization actually happens, though.

Reply:
Sure, we have meanwhile another change in the pipeline, more precision of ballast to cover this problem ( https://github.com/iltis42/XCVario/issues/74 ), that we needed to address as well in XCS/XCVario.cpp, so i will move as well to GetHectoPascal() in here of course.